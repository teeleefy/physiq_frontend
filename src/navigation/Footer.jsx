
import "./styles/Footer.css";
import { NavLink } from "react-router-dom";
import { Navbar, Nav, NavItem } from "reactstrap";


/** Navigation bar for site. Shows up on every page.
 *
 * When user is logged in, shows links to main areas of site. When not,
 * shows link to Login and Signup forms.
 *
 * Rendered by App.
 */

function Footer() {
  return (
    <div>
        
                    <NavLink id="Footer-linkedin" className="Footer-navlink" to="https://tammypainter.carrd.co/"><i className="fa-solid fa-user"></i></NavLink>
                    <NavLink id="Footer-linkedin" className="Footer-navlink" to="https://www.linkedin.com/in/tammy-painter-73416b282/"><i className="fa-brands fa-linkedin"></i></NavLink>
                    <NavLink  id="Footer-github" className="Footer-navlink" to="https://github.com/teeleefy"><i className="fa-brands fa-github"></i></NavLink>
                    <NavLink id="Footer-envelope" className="Footer-navlink" to="mailto:tammypainter.codes@gmail.com"><i className="fa-regular fa-envelope"></i></NavLink>
    </div>
  );
}

export default Footer;
