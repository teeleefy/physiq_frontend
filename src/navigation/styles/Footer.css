#Footer-nav{
  text-align: center;
  justify-content: center;
  margin: auto;

}

.Footer-navlink{
    padding: 0 2rem 0 2rem;
    font-size: 2rem;
    color: red;
    text-shadow: 4px 4px 3px black;
}

.Footer-navlink:hover {
    color: black;
    text-shadow: 3px 3px 5px red;
  }