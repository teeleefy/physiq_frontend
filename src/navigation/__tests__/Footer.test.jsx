import { renderWithRouter, screen } from '../../../__tests__/test-tools';
import Footer from '../Footer'

describe('Footer', () => {
  it('renders the Footer component', () => {
    renderWithRouter(<Footer />)
    screen.debug(); // prints out the jsx in the Footer component unto the command line
  })
})

it("matches snapshot", function() {
    const { asFragment } = renderWithRouter(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });

