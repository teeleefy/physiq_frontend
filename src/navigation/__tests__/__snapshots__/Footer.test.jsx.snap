// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <a
      class="Footer-navlink"
      href="https://tammypainter.carrd.co/"
      id="Footer-linkedin"
    >
      <i
        class="fa-solid fa-user"
      />
    </a>
    <a
      class="Footer-navlink"
      href="https://www.linkedin.com/in/tammy-painter-73416b282/"
      id="Footer-linkedin"
    >
      <i
        class="fa-brands fa-linkedin"
      />
    </a>
    <a
      class="Footer-navlink"
      href="https://github.com/teeleefy"
      id="Footer-github"
    >
      <i
        class="fa-brands fa-github"
      />
    </a>
    <a
      class="Footer-navlink"
      href="mailto:tammypainter.codes@gmail.com"
      id="Footer-envelope"
    >
      <i
        class="fa-regular fa-envelope"
      />
    </a>
    <a
      aria-current="page"
      class="active"
      href="/"
      id="Footer-copyright"
    >
      Ⓒ 
      <span
        id="Footer-rights"
      >
         2024   family chart    All rights reserved
      </span>
    </a>
  </div>
</DocumentFragment>
`;
